src/create-reminder-signup/lambda/lambda.ts

```diff
@@ -97,6 +98,34 @@ const runRecurring = async (signupRequest: unknown): Promise<void> => {
 	return createSignup(signupRequest, recurringSignupRequestSchema, persist);
 };
 
+/**
+ * There are some validation errors that currently end up on the DLQ which we can't actually fix.  This ignores those.
+ * @param identityResult
+ * @returns Promise<void>
+ */
+const ignoreSomeValidationErrors = async (
+	identityResult: IdentityFailure,
+): Promise<void> => {
+	// blocked email providers should not be passed to DLQ
+	const ignoreBlockedProvider = identityResult.errorMessage?.errors.filter(
+		(e) => {
+			e.message === 'Registration Error';
+		},
+	);
+	// email addresses that fail validation should not be passed to the DQL
+	const ignoreInvalidEmailAddress =
+		identityResult.errorMessage?.errors.filter((e) => {
+			e.description === 'Invalid email format' ||
+				e.message == 'Invalid emailAddress:';
+		});
+
+	if (!(ignoreBlockedProvider || ignoreInvalidEmailAddress)) {
```

Comment:
I'm struggling a bit with the logic in this function.
The type of these values is `IdentityErrorMessage[] | undefined`. If the errors array is defined but does not contain one of these errors, then I think we end up with an empty array. In that case I'm not sure this is the behaviour that we want.
Perhaps a clearer way of doing this would be to use Array.some instead, since that gives us a boolean, e.g.
```
const shouldIgnoreError = identityResult.errorMessage?.errors.some(
	(error) =>
		error.message === 'Registration Error' ||
		error.message === 'Invalid emailAddress:',
);

if (shouldIgnoreError) {
...
```